src/Enzo/enzo_EnzoReconstructor.hpp

```diff
@@ -59,32 +55,40 @@ class EnzoReconstructor : public PUP::able
-  /// @param priml_group,primr_group holds field names where the reconstructed
-  ///     left/right face-centered primitives will be stored. The relevant
-  ///     fields should be formally defined as cell-centered (to allow for
-  ///     reuse along multiple dimensions). During the calculation, they are
-  ///     treated as face-centered (without having values on the exterior faces
-  ///     of the block). Consequentially there will be some unused space at the
-  ///     end of the arrays.  
-  /// @param dim Dimension along which to reconstruct interface values. Values
-  ///     of 0, 1, and 2 correspond to the x, y, and z directions, respectively.
-  /// @param eos Pointer to an instance of EnzoEquationOfState object
-  /// @param stale_depth indicates the current stale_depth for the supplied
-  ///     cell-centered quantities
-  virtual void reconstruct_interface (Block *block, Grouping &prim_group,
-				      Grouping &priml_group,
-				      Grouping &primr_group, int dim,
-				      EnzoEquationOfState *eos,
-				      int stale_depth)=0;
+  /// @param[in]  prim_map Map holding the data for the cell-centered
+  ///     reconstructable primitives. This is expected to have keys for all
+  ///     of the active reconstructed quantities registered with the factory
+  ///     method (plus all of the keys listed in `passive lists`)
+  /// @param[out] priml_map,primr_map Holds existing arrays where the
+  ///     left/right reconstructed, face-centered primitives are written.
+  ///     These must supply the same keys that are expected for prim_map.
+  ///     The arrays are expected to have identical shapes to those in
+  ///     prim_map, except along dimension `dim`. Along that dimension, these
+  ///     arrays should hold one fewer value.
+  /// @param[in]  dim Dimension along which to reconstruct interface values.
+  ///     Values of 0, 1, and 2 correspond to the x, y, and z directions.
+  /// @param[in]  eos Pointer to an instance of EnzoEquationOfState object
+  /// @param[in]  stale_depth indicates the current stale_depth for the
+  ///     supplied cell-centered quantities
+  /// @param[in]  passive_lists A list of lists of keys for passive scalars. In
+  ///     this method, this is effectively concatenated into one list of keys.
+  ///
+  /// @note It's alright for arrays in `priml_map` and `primr_map` to have the
+  /// shapes of cell-centered arrays (i.e. the same shape as arrays in
+  /// `prim_map`). In this case, the function effectively treats such arrays as
+  /// if their `subarray` method were invoked, where `CSlice(0,-1)` is
+  /// specified for the `dim` axis and `CSlice(nullptr,nullptr)` is specified
+  /// for other axes.
+  virtual void reconstruct_interface
+  (EnzoEFltArrayMap &prim_map, EnzoEFltArrayMap &priml_map,
+   EnzoEFltArrayMap &primr_map, int dim, EnzoEquationOfState *eos,
+   int stale_depth,
+   const std::vector<std::vector<std::string>>& passive_lists)=0;
```

Comment:
I was just about to suggest replacing `std::vector<std::string>` with `str_vec_t` again and am now wondering if it'd be worth to introduce a `passive_list_t` of `std::vector<std::vector<std::string>>`. What do you think?